File: DPmenu/paiza8.py
n, x, a, y, b, z, c = map(int, input().split())

dp = [100000] * (n + z)

dp[0] = 0

for i in range(x, n+z):
   if i - x >= 0:
        dp[i] = min(dp[i], dp[i - x] + a)
   if i - y >= 0:
        dp[i] = min(dp[i], dp[i - y] + b)
   if i - z >= 0:
        dp[i] = min(dp[i], dp[i - z] + c)
print(min(dp[n:]))

File: Brank/paiza8.swift
let n = Int(readLine()!)! 
var dict = [String: Int]() 

for _ in 0..<n {
    let input = readLine()!.split(separator: " ") 
    let s = String(input[0]) 
    let d = Int(input[1])! 
    dict[s, default: 0] += d
}

let sortedDict = dict.sorted { $0.value > $1.value } 
for (key, value) in sortedDict {
    print("\(key) \(value)") 
}